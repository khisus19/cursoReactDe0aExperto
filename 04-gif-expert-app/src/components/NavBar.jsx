
import Logo from "../assets/logo-long-white.svg";

export const NavBar = () => {
  return (
    <nav>
      <div>
          <a href="index.html" id="logo">
              <img src={ Logo } alt="Khisus19 logo" />
          </a>
      </div>
      <ul class="nav-link-ul">
          <li><a href = "mailto: khisus19@gmail.com">Contact me</a></li>
          <li><a href="https://khisus19-dev.netlify.app/" target="_blank">My Web site</a></li>
      </ul>
    </nav>
  )
}
