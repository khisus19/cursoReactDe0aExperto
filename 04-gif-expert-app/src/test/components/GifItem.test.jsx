import { render, screen } from '@testing-library/react'; // Import de la funciÃ³n render
import { GifItem } from '../../components/GifItem'; // Import del componente


describe('Tests for GifItem Component', () => {
  
  const title = "Stranger things";
  const url   = 'https://one-punch.com/saitama.jpg';

  test('Should match the snapshot', () => { 
    
    const { container } = render( <GifItem title={ title } url={ url } /> );
    expect( container ).toMatchSnapshot();
  });

  test('Debe mostrar la imagen con el URL y el ALT indicado', () => { 
    
    render( <GifItem title={ title } url={ url } /> );
    // screen.debug();
    // expect( screen.getByRole('img').src ).toBe( url );
    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe( url );
    expect( alt ).toBe( title );

  });

  test('Debe mostrar el titulo en el componente', () => { 
    
    render( <GifItem title={ title } url={ url } /> );
    expect( screen.getByText( title ) ).toBeTruthy();

  })
});