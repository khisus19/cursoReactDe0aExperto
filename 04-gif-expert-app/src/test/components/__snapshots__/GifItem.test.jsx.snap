// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifItem Component Should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Stranger things"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Stranger things
    </p>
  </div>
</div>
`;
